data/topic-filter.go

```diff
@@ -0,0 +1,222 @@
+package data
+
+import (
+	"context"
+	"net/url"
+	"strings"
+
+	errs "github.com/ONSdigital/dp-frontend-search-controller/apperrors"
+	"github.com/ONSdigital/log.go/v2/log"
+)
+
+// Filter represents information of filters selected by user
+type TopicFilter struct {
+	Query           []string `json:"query,omitempty"`
+	LocaliseKeyName []string `json:"localise_key,omitempty"`
+}
+
+// TopicCategory represents all the search categories in search page
+type TopicCategory struct {
+	LocaliseKeyName string             `json:"localise_key"`
+	Count           int                `json:"count"`
+	ContentTypes    []TopicContentType `json:"content_types"`
+}
+
+// ContentType represents the type of the search results and the number of results for each type
+type TopicContentType struct {
+	LocaliseKeyName string   `json:"localise_key"`
+	Count           int      `json:"count"`
+	Group           string   `json:"group"`
+	Types           []string `json:"types"`
```

Comment:
Is this field just the subtopics? So would it not be better to rename it so?

```suggestion
	SubTopics           []string `json:"subtopics"`
```